src/main/java/us/abstracta/wiresham/VirtualTcpClient.java

```diff
@@ -58,21 +59,38 @@ private Socket buildSocket()
   }
 
   public void run() {
-    try {
-      new ConnectionFlowDriver(buildSocket(), readBufferSize, flow).run();
-    } catch (NoSuchAlgorithmException | KeyManagementException | IOException e) {
-      LOG.error("Problem starting client", e);
-    }
-  }
-
-  public void start() throws IOException, KeyManagementException, NoSuchAlgorithmException {
-    connection = new ConnectionFlowDriver(buildSocket(), readBufferSize, flow);
     executorService = Executors.newSingleThreadExecutor();
+    connection = new ConnectionFlowDriver(buildFlowConnectionProvider(), flow, port);
     executorService.submit(connection);
   }
 
-  public void stop(long timeoutMillis) throws IOException, InterruptedException {
-    connection.close();
+  private FlowConnectionProvider buildFlowConnectionProvider() {
+    return new FlowConnectionProvider() {
+
+      public final Map<Integer, FlowConnection> map = new ConcurrentHashMap<>();
+
+      @Override
+      public FlowConnection get(int port) throws IOException {
+        map.putIfAbsent(port, new FlowConnection(buildSocket(port), readBufferSize));
```

Comment:
this will create a new socket on every invocation, you need the form that receives a supplier as parameter to avoid creating sockets when there is already one

Reply:
I think the method is compute, and it returns the created element or the existing one